src/devices/Bmxx80/Bmx280Base.cs

```diff
@@ -272,34 +270,31 @@ protected override void SetDefaultConfiguration()
         }
 
         /// <summary>
-        /// Compensates the pressure in Pa, in Q24.8 format (24 integer bits and 8 fractional bits).
+        /// Compensates the pressure in Pa, in double format
         /// </summary>
         /// <param name="adcPressure">The pressure value read from the device.</param>
-        /// <returns>Pressure in Hectopascals (hPa).</returns>
-        /// <remarks>
-        /// Output value of “24674867” represents 24674867/256 = 96386.2 Pa = 963.862 hPa.
-        /// </remarks>
+        /// <returns>Pressure as an instance of <see cref="Pressure"/>.</returns>
         private Pressure CompensatePressure(long adcPressure)
         {
             // Formula from the datasheet http://www.adafruit.com/datasheets/BST-BMP280-DS001-11.pdf
-            // The pressure is calculated using the compensation formula in the BMP280 datasheet
-            long var1 = TemperatureFine - 128000;
-            long var2 = var1 * var1 * (long)_calibrationData.DigP6;
-            var2 = var2 + ((var1 * (long)_calibrationData.DigP5) << 17);
-            var2 = var2 + ((long)_calibrationData.DigP4 << 35);
-            var1 = ((var1 * var1 * (long)_calibrationData.DigP3) >> 8) + ((var1 * (long)_calibrationData.DigP2) << 12);
-            var1 = ((((1L << 47) + var1)) * (long)_calibrationData.DigP1) >> 33;
-            if (var1 == 0)
+            // This uses the recommended approach with floating point math
+            double var1, var2, p;
+            var1 = (TemperatureFine / 2.0) - 64000.0;
+            var2 = var1 * var1 * ((double)_calibrationData.DigP6) / 32768.0;
+            var2 = var2 + var1 * ((double)_calibrationData.DigP5) * 2.0;
+            var2 = (var2 / 4.0) + (((double)_calibrationData.DigP4) * 65536.0);
+            var1 = (((double)_calibrationData.DigP3) * var1 * var1 / 524288.0 + ((double)_calibrationData.DigP2) * var1) / 524288.0;
+            var1 = (1.0 + var1 / 32768.0) * ((double)_calibrationData.DigP1);
+            if (var1 == 0.0)
```

Comment:
can we still hit the case where this is 0?

Reply:
Unlikely, but the function is taken directly from the spec sheet, and there this condition is used.